File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/registry/RecordData.java
package com.esferalia.aon.occam.api.model.registry;

import java.io.Serializable;
import java.util.Date;

public class RecordData implements Serializable{
	
	private static final long serialVersionUID = 9114564405091033572L;
	
	private Integer id;
	private Integer domain;
	private Integer registry;
	private Date creationDate;
	private String description;
	private String notary;
	private String number;
	private Date recordDate;
	private String volume;
	private String section;
	private String page;
	private String sheet;
	private String registration;	
	private Integer attach;
	
	public Integer getId() {
		return id;
	}
	public RecordData setId(Integer id) {
		this.id = id;
		return this;
	}
	public Integer getDomain() {
		return domain;
	}
	public RecordData setDomain(Integer domain) {
		this.domain = domain;
		return this;
	}
	public Integer getRegistry() {
		return registry;
	}
	public RecordData setRegistry(Integer registry) {
		this.registry = registry;
		return this;
	}
	public Date getCreationDate() {
		return creationDate;
	}
	public RecordData setCreationDate(Date creationDate) {
		this.creationDate = creationDate;
		return this;
	}
	public String getDescription() {
		return description;
	}
	public RecordData setDescription(String description) {
		this.description = description;
		return this;
	}
	public String getNotary() {
		return notary;
	}
	public RecordData setNotary(String notary) {
		this.notary = notary;
		return this;
	}
	public String getNumber() {
		return number;
	}
	public RecordData setNumber(String number) {
		this.number = number;
		return this;
	}
	public Date getRecordDate() {
		return recordDate;
	}
	public RecordData setRecordDate(Date recordDate) {
		this.recordDate = recordDate;
		return this;
	}
	public String getVolume() {
		return volume;
	}
	public RecordData setVolume(String volume) {
		this.volume = volume;
		return this;
	}
	public String getSection() {
		return section;
	}
	public RecordData setSection(String section) {
		this.section = section;
		return this;
	}
	public String getPage() {
		return page;
	}
	public RecordData setPage(String page) {
		this.page = page;
		return this;
	}
	public String getSheet() {
		return sheet;
	}
	public RecordData setSheet(String sheet) {
		this.sheet = sheet;
		return this;
	}
	public String getRegistration() {
		return registration;
	}
	public RecordData setRegistration(String registration) {
		this.registration = registration;
		return this;
	}
	public Integer getAttach() {
		return attach;
	}
	public RecordData setAttach(Integer attach) {
		this.attach = attach;
		return this;
	}
}

File: aon-solutions/src/main/webapp/models/registry/RecordData.js
import * as LS from '../../services/localStorageService.js';

export class RecordData {
    id;
    domain;
    registry;
	creationDate;
	description;
	notary;
	number;
	recordDate;
	volume;
	section;
	page;
	sheet;
	registration;	
	attach;
    
    constructor(recordData) {
        if(recordData) {
            this.id = recordData.id;
            this.domain = recordData.domain || LS.getDomainId();
            this.registry = recordData.registry;
            this.creationDate = recordData.creationDate;
            this.description = recordData.description || 'Escritura de constitución';
            this.notary = recordData.notary || '';
            this.number = recordData.number || '';
            this.recordDate = recordData.recordDate;
            this.volume = recordData.volume || '';
            this.section = recordData.section || '';
            this.page = recordData.page || '';
            this.sheet = recordData.sheet || '';
            this.registration = recordData.registration || 'Inscrita en el Registro mercantil';	
            this.attach = recordData.attach;
        } else {
            this.description = 'Escritura de constitución';
            this.registration = 'Inscrita en el Registro mercantil';
            this.volume = '';
            this.section = '';
            this.page = '';
            this.sheet = '';
        }
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id; 
        return this;
    }

    getDomain() {
        return this.domain;
    }

    setDomain(domain) {
        this.domain = domain; 
        return this;
    }

    getRegistry(){
        return this.registry;
    }
    
    setRegistry(registry){ 
        this.registry = registry;
        return this;
    }

    getCreationDate() {
        return this.creationDate;
    }

    setCreationDate(creationDate) {
        this.creationDate = creationDate;
        return this;
    }

    getRecordDate() {
        return this.recordDate;
    }

    setRecordDate(recordDate) {
        this.recordDate = recordDate;
        return this;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
        return this.description;
    }

    getNotary() {
        return this.notary;
    }

    setNotary(notary) {
        this.notary = notary;
        return notary;
    }

    getNumber() {
        return this.number;
    }

    setNumber(number) {
        this.number = number;
        return this;
    }

    getVolume() {
        return this.volume;
    }

    setVolume(volume) {
        this.volume = volume;
        return this;
    }

    getSection() {
        return this.section;
    }

    setSection(section) {
        this.section = section;
        return this;
    }
    
    getPage() {
        return this.page;
    }

    setPage(page) {
        this.page = page;
        return this;
    }
    
    getSheet() {
        return this.sheet;
    }

    setSheet(sheet) {
        this.sheet = sheet;
        return this;
    }

    getRegistration() {
        return this.registration;
    }	

    setRegistration(registration) {
        this.registration = registration;
        return this;
    }

    getAttach() {
        return this.attach;
    }

    setAttach(attach) {
        this.attach = attach;
        return this; 
    }

}